# Suggested Reading

Below are some specific reading suggestions, but in general the topics you'll want to read around are:

- zebrafish
- Illumina sequencing
- exome sequencing
- short read alignment (e.g. BWA)
- variant calling (e.g. freebayes, GATK, VCF, GVCF)
- workflow management (e.g. Snakemake)

If there are some papers you can't access then please let us know.

Some of the suggestions below aren't references you'd want to cite in your thesis (e.g. Wikipedia articles), but they do provide useful background.

## Zebrafish

- [What are model organisms?](https://www.yourgenome.org/facts/what-are-model-organisms)
- [Why use the zebrafish in research?](https://www.yourgenome.org/facts/why-use-the-zebrafish-in-research)
- [Tiny fish, big splash: the story of the zebrafish](https://www.yourgenome.org/stories/tiny-fish-big-splash-the-story-of-the-zebrafish)
- [Zebrafish earns its stripes](https://www.nature.com/articles/nature12094)
- [The zebrafish reference genome sequence and its relationship to the human genome](https://www.nature.com/articles/nature12111)
- [A systematic genome-wide analysis of zebrafish protein-coding gene function](https://www.nature.com/articles/nature11992) - Zebrafish Mutation Project
- [ZFIN](http://zfin.org/) - the Zebrafish Information Network
- [Ensembl Zebrafish](https://www.ensembl.org/Danio_rerio/) - Ensembl genome browser for Danio rerio

## Sequencing

- [What is the Illumina method of DNA sequencing?](https://www.yourgenome.org/facts/what-is-the-illumina-method-of-dna-sequencing)
- [An introduction to Next-Generation Sequencing Technology](https://emea.illumina.com/content/dam/illumina-marketing/documents/products/illumina_sequencing_introduction.pdf)
- [Exome Sequencing: Current and Future Perspectives](https://academic.oup.com/g3journal/article/5/8/1543/6025359)

## Alignment

- [Next Generation Sequencing (NGS)/Alignment](https://en.wikibooks.org/wiki/Next_Generation_Sequencing_(NGS)/Alignment)
- [BWA](https://github.com/lh3/bwa)
- [Fast and accurate short read alignment with Burrowsâ€“Wheeler transform](https://academic.oup.com/bioinformatics/article/25/14/1754/225615)

## Variant Calling

- [Next Generation Sequencing (NGS)/DNA Variants](https://en.wikibooks.org/wiki/Next_Generation_Sequencing_(NGS)/DNA_Variants)
- [Genotype and SNP calling from next-generation sequencing data](https://www.nature.com/articles/nrg2986)
- [SAMtools & BCFtools](https://www.htslib.org/)
- [Twelve years of SAMtools and BCFtools](https://academic.oup.com/gigascience/article/10/2/giab008/6137722)
- [freebayes](https://github.com/freebayes/freebayes)
- [Haplotype-based variant detection from short-read sequencing](https://arxiv.org/abs/1207.3907)
- [GATK](https://gatk.broadinstitute.org/)
- [Germline short variant discovery (SNPs + Indels)](https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels-)
- [VCF format](https://en.wikipedia.org/wiki/Variant_Call_Format)
- [GVCF format](https://gatk.broadinstitute.org/hc/en-us/articles/360035531812-GVCF-Genomic-Variant-Call-Format)
- [DeepVariant](https://github.com/google/deepvariant)

## Workflow Management

- [Snakemake](https://snakemake.github.io/)
- [Snakemake documentation](https://snakemake.readthedocs.io/en/stable/)
- [Sustainable data analysis with Snakemake](https://f1000research.com/articles/10-33/v2)
- [Example Snakemake variant calling workflow](https://github.com/snakemake-workflows/dna-seq-gatk-variant-calling)

## Apocrita

- [QMUL HPC documentation](https://docs.hpc.qmul.ac.uk/)
- [ITS Research courses](https://learn.hpc.qmul.ac.uk/)
